Game-Inventory.API/Controllers/CompaniesController.cs

```diff
@@ -0,0 +1,33 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using System.Runtime.InteropServices;
+
+namespace Game_Inventory.API.Controllers
+{
+    public class CompaniesController
+    {
+
+        [HttpGet]
+        public ExternalException GetCompanies(string name)
+        {
+            return new ExternalException();
+        }
+
+        [HttpDelete]
+        public ExternalException DeleteCompanies(string name)
+        {
+            return new ExternalException();
+        }
+
+        [HttpPost]
+        public ExternalException CreateCompanies(string name)
+        {
+            return new ExternalException();
+        }
+
+        [HttpPut]
+        public ExternalException PutCompanies(string name)
```

Comment:
Revisar el objecot que estas devolviendo, usualmente los controladdores devuelven un IAction o IActionResult no se retornan las excepciones, sino que se las lanzan. Cambiar por throw Not implemented exception()

Reply:
La nomenclatura de los put suele ser la siguiente /companies/{id} porque le estamos indicando que actualize todos los registros del id especificado. Los datos a modificar van en el body, por lo que este metodo tiene que tener dos parametros uno que sea el de la ruta y otro que sea del body, para simplificar puedes ponerle un string en vez de ponerle una estructura para el body